core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -629,12 +630,15 @@ class KafkaApis(val requestChannel: RequestChannel,
       AdminUtils.createTopic(zkUtils, topic, numPartitions, replicationFactor, properties, RackAwareMode.Safe)
       info("Auto creation of topic %s with %d partitions and replication factor %d is successful"
         .format(topic, numPartitions, replicationFactor))
-      new MetadataResponse.TopicMetadata(Errors.LEADER_NOT_AVAILABLE, topic, java.util.Collections.emptyList())
+      new MetadataResponse.TopicMetadata(Errors.LEADER_NOT_AVAILABLE, topic, common.Topic.isInternal(topic),
```

Comment:
This is related to this discussion too: https://github.com/apache/kafka/pull/1095#discussion_r60764126

Right now some these classes have quite a bit of logic and don't expose the "raw" information. I agree that it would be nice to have some layering here where the logic is handled/exposed elsewhere. I don't want to impose to much change on existing logic in this patch though either.